### PR TITLE
Arc-Electroplating-Support-For-Borgs

### PR DESCRIPTION
**Pull request by Francisco-Franks**
[Silicon][Materials]

## About the PR 
This enables Borgs, and well, anyone tangible, to put the correct things into the arc electroplater. Click drag things into the plater.

## Why's this needed? 
Currently the electroplater seems rather seldom used. This will let borgs join in on the fun, maybe we will see more soulsteel orchestras or something.

## Changelog 

```changelog
(u)Metareferencer
(+)Click-dragging now works for the electroplater. Get electroplating, borgos!
```

### CODE REVIEW

File: code/obj/machinery/arc_electroplater.dm [hunk 1]
```diff
@@ -45,6 +45,37 @@ TYPEINFO(/obj/machinery/arc_electroplater)
...
+			boutput(user, SPAN_ALERT("There is no way that could fit!"))
+			return
+		if (W.cant_drop) //For borg held items
+			boutput(user, SPAN_ALERT("You can't put that in [src] when it's attached to you!"))
+			return
+		if (iswraith(user) || isintangible(user) || is_incapacitated(user)|| isghostdrone(user) || isAI(user))
+			boutput(user, SPAN_ALERT("Electroplating is for the Tangible."))
+			return
+		else
+			src.Attackby(W, user)
```
Comment (pgmzeta) [inline code/obj/machinery/arc_electroplater.dm:77]: calling attackby results in some duplicate checks; I would pull the common stuff into item level procs and call those from both to ensure the logic is mirrored between clickdrag and attackby now and in the future.

Comment (Francisco-Franks) [reply code/obj/machinery/arc_electroplater.dm]: Ok, attackby and mousedrop_t now just route directly to a proc which does it all.

File: code/obj/machinery/arc_electroplater.dm [hunk 2]
```diff
@@ -45,8 +45,31 @@ TYPEINFO(/obj/machinery/arc_electroplater)
...
+	// See if I can piece together how to make this fly.
+	proc/PlaterInteract(var/obj/item/W, var/mob/user)
+		// Please don't drag the nano-crucible into the plater. Or any other machine or mob for that matter.
+		if (!istype(W, /obj/item))
+			boutput(user, SPAN_ALERT("Please do not attempt to plate machinery or entities."))
+			return
+		// Theres a suicide button for this.
+		if (W == user)
+			boutput(user, SPAN_ALERT("Please do not drag yourself into the Plater."))
+			return
```
Comment (Tarmunora) [inline code/obj/machinery/arc_electroplater.dm:64]: These could honestly just fail silently

File: code/obj/machinery/arc_electroplater.dm [hunk 3]
```diff
@@ -45,8 +45,31 @@ TYPEINFO(/obj/machinery/arc_electroplater)
...
+		// Do not attempt to plate objects from a distance.
+		if (BOUNDS_DIST(user, src) > 0)
+			return
+		// Do not attempt to plate objects at a distance.
+		if (BOUNDS_DIST(W, src) > 0)
+			return
```
Comment (Tarmunora) [inline code/obj/machinery/arc_electroplater.dm:70]: Might also want to check distance between `user` and `W`, so you can't pull something into the plater from the far side of it. Minor thing but good practice

File: code/obj/machinery/arc_electroplater.dm [hunk 4]
```diff
@@ -91,7 +114,7 @@ TYPEINFO(/obj/machinery/arc_electroplater)
 			boutput(user, SPAN_ALERT("You can't plate something this tiny!"))
 			return
 
-		if (W.w_class > src.max_wclass || istype(W, /obj/item/storage/secure))
+		if (W.w_class > src.max_wclass || istype(W, /obj/item/storage/secure) || W.anchored >= ANCHORED)
```
Comment (Tarmunora) [inline code/obj/machinery/arc_electroplater.dm:117]:
```suggestion
		if (W.w_class > src.max_wclass || istype(W, /obj/item/storage/secure) || W.anchored)
```
Unless something has a negative (or fractional, I suppose) value for `anchored`, the check can be simplified (and usually is)

**Review by Tarmunora — CHANGES_REQUESTED** [review]

**Review by Francisco-Franks — COMMENTED** [review]
Agreedo

**Review by Tarmunora — APPROVED** [review]

**Review by pgmzeta — APPROVED** [review]

**Pull request merged**